config/config.go

```diff
@@ -80,6 +80,8 @@ type Configuration struct {
 	RequestValidation RequestValidation `mapstructure:"request_validation"`
 	// When true, PBS will assign a randomly generated UUID to req.Source.TID if it is empty
 	AutoGenSourceTID bool `mapstructure:"auto_gen_source_tid"`
+	//When true, new bid id will be generated in seatbid[].bid[].ext.prebid.bidid and used in event urls instead
+	GenerateBidID bool `mapstructure:"generate_bid_id"`
```

Comment:
Good question. The answer could point to the best place for the config option.

I can see it as a feature independent from events. I remember @bretg mentioning how prebid.js looks for it to sub the original bid id.

This implementation does appear co-dependent on the events code, but that could be to avoid another nested iteration over the bids - it doesn't actually need events to be enabled. Another option to separate it could be at bid collection time - actually could be nice if this could be also controllable by bidder.

Reply:
It's now decoupled from events, please see previous comment for details. 
I'm confused. Do you mean I need to overwrite existing bidId returned from bidder with generated bidId?

Reply:
No need to overwrite, sorry for the misleading link. I had added it to show how prebid.js uses the ext.prebid.bidid field.

Reply:
I'm not sure how it's implemented in PBS-Java. Based on this line in the spec:

> Prebid Server gains a configuration that will cause it to generate a unique UUID for each successful bid and place it in seatbid[].bid[].ext.prebid.bidid.

I believe we should assign the value in the bid collection loop. The `modifyBidsForEvents` should read the `prebid.bidid` value which would've been already populated.

> actually could be nice if this could be also controllable by bidder.

Could be. Sure. Let's keep this PR focused on a global host config and bring the per-bidder config up with the Prebid Server Committee if you think it has value (certainly may).

Reply:
Refactored